pkg/testing/fixture.go

```diff
@@ -335,16 +335,30 @@ func (f *Fixture) Exec(ctx context.Context, args []string, opts ...process.CmdOp
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
+	cmd, err := f.PrepareAgentCommand(ctx, args, opts...)
+	if err != nil {
+		return nil, fmt.Errorf("error creating cmd: %w", err)
+	}
+	f.t.Logf(">> running agent with: %v", cmd.Args)
+
+	return cmd.CombinedOutput()
+}
+
+// PrepareAgentCommand creates an exec.Cmd ready to execute an elastic-agent command.
+func (f *Fixture) PrepareAgentCommand(ctx context.Context, args []string, opts ...process.CmdOption) (*exec.Cmd, error) {
+	err := f.ensurePrepared(ctx)
```

Comment:
Looks like this gets called twice in `fixture.Exec` now, once at the beginning of the function and again here.

Reply:
It's a function that is safe to call multiple times (if the fixture has been already prepared, it does nothing).
I prefixed like that in case someone has to prepare an agent command and didn't call the Prepare() function beforehand.
The Prepare() function is needed here as we rely on the setting of `workDir` that is part of `binaryPath` to operate in the correct extracted folder for the respective test.
It is a bit convoluted but it boils down to being safe is someone does something like:
```
f := atesting.NewFixture(...)

cmd := f.PrepareAgentCommand(...)
stderr := cmd.StderrPipe()
...
cmd.Run()
```

The need for this function comes from the fact that Exec() will return the combined stdout and stderr bytes while, in order to test the correct unmarshalling of stdout and the error message in stderr for `elastic-agent version` command I need to keep them separate.